### PR TITLE
Player Blocker (Requested)

### PR DESCRIPTION
**Pull request by NekoRem**
Most people had trouble with making/pasting a block bot so here is mine. Was done in 5 minutes and isn't smooth at all. In the future i may make another pull request with a better one.

### CODE REVIEW

**Comment by Evilhelder** [comment]
It is not working, I tried to fix it with a different code but without success ..

**Comment by Evilhelder** [comment]
I fixed add this line of code "Misc::playerBlocker(cmd);" in the Hook.cpp file...

image

**Comment by NekoRem** [comment]
> I fixed add this line of code "Misc::playerBlocker(cmd);" in the Hook.cpp file...
> 
> image

ah yeah sorry wasn't really paying attention... and completely forgot

**Comment by Evilhelder** [comment]
But, he is not respecting the on button, if I leave the trigger checked the blockbot is activated all the time, if I just add the button on the trigger without turning it on, it will only activate with the button.


https://youtu.be/SJu3uKuyIv4

**Comment by Evilhelder** [comment]
Even so, it is working very well.

**Comment by NekoRem** [comment]
> But, he is not respecting the on button, if I leave the trigger checked the blockbot is activated all the time, if I just add the button on the trigger without turning it on, it will only activate with the button.
> 
> https://youtu.be/SJu3uKuyIv4

replace || with && in the if statement within the playerblocker function

**Comment by Evilhelder** [comment]
> replace || with && in the if statement within the playerblocker function

thanks.

Now the next step is to do a RunBoostHelper ....

I'll take a code from another cheat and try to translate it to osiris, if I get any results I'll post it here in case you want to add the pull request ...

**Comment by NekoRem** [comment]
for a run boost you can just add forward and backward move to this i believe. it would also make a head standing blockbot

**Comment by xAkiraMiura** [comment]
Does this respect the direction that the target is facing or does it always try to block them in 1 cardinal direction.

**Comment by NekoRem** [comment]
> Does this respect the direction that the target is facing or does it always try to block them in 1 cardinal direction.

1 cardinal direction.

**Comment by notgoodusename** [comment]
bruh i already have done this and the distance func can be replaced with something more easy to understand and use, and the calcangle can be done like the aimbot calcangle, you should copy the target movement by doing target->velocity().lenght() (2d is almost the same for my personal opinion) and multiplying it by 1.8 and setting it to sidemove

**Comment by Evilhelder** [comment]
Someone help me with this code?


void Misc::runboostbot(UserCmd* cmd) noexcept
    {
        public static bool boosting = false;
        public static void Start() //disabled
        {
            while (true)
            {
                Thread.Sleep(1);

                if (!Properties.Settings.Default.runboostbot)
                    continue;

                if (!LocalPlayer.InGame)
                    continue;

                if (LocalPlayer.Health <= 0)
                    continue;

                if (LocalPlayer.Dormant)
                    continue;

                Entity target = null;

                while ((DllImport.GetAsyncKeyState(Properties.Hotkey.Default.runboostbotKey) & 0x8000) != 0)
                {
                    Thread.Sleep(1);

                    if (target == null)
                    {
                        target = Aimbot.BestDistance();
                    }

                    boosting = true;

                    Vector3 position = target.Position;

                    if (target.Speed <= 0)
                    {
                        LocalPlayer.MoveClearX();
                    }
                    else
                    {
                        LocalPlayer.ViewAngleY = Aimbot.NormalizedAngle(Aimbot.CalcAngle(LocalPlayer.ViewPosition, position)).y;

                        LocalPlayer.MoveForward();
                    }
                    
                }

                if (boosting)
                {
                    LocalPlayer.MoveClearX();
                    boosting = false;
                }

**Comment by Laxybaxy** [comment]
why not do this?
then you wont have to make a new distancefunc 
float dist = (Vector)(entity->origin()-localPlayer->origin()).lenght();

**Comment by notgoodusename** [comment]
> why not do this?
> then you wont have to make a new distancefunc
> float dist = (Vector)(entity->origin()-localPlayer->origin()).lenght();

because he took it from my old pasted code

**Comment by Laxybaxy** [comment]
i see

File: Osiris/Hacks/Misc.cpp [hunk 1]
```diff
@@ -66,6 +66,51 @@ void Misc::slowwalk(UserCmd* cmd) noexcept
...
+        for (int i = 1; i <= interfaces.engine->getMaxClients(); i++) {
+            auto entity = interfaces.entityList->getEntity(i);
+
+            if (!entity)
+                continue;
+
+            if (!entity->isAlive() || entity->isDormant() || entity == localPlayer)
+                continue;
+
+            float dist = distance(localPlayer->origin(), entity->origin());
```
Comment (Laxybaxy) [inline Osiris/Hacks/Misc.cpp]: use (Vector)(localPlayer->origin() - entity->origin()).lenght();

File: Osiris/Hacks/Misc.cpp [hunk 2]
```diff
@@ -66,6 +66,51 @@ void Misc::slowwalk(UserCmd* cmd) noexcept
...
+        if (!target)
+            return;
+
+        Vector angles = Misc::calculateRelativeAngle(localPlayer->origin(), target->origin());
+
+        angles.y -= localPlayer->eyeAngles().y;
+        angles.normalize();
+        angles.y = std::clamp(angles.y, -180.f, 180.f);
+
+        if (angles.y < 0.0f)
```
Comment (Laxybaxy) [inline Osiris/Hacks/Misc.cpp]: make a treshhold to remove jiggeling ex `if (angles.y < 1.0f)` and the other one `if (angles.y > 1.0f)`, needs to be tuned tho

File: Osiris/Hacks/Misc.cpp [hunk 3]
```diff
@@ -506,3 +551,24 @@ void Misc::killMessage(GameEvent& event) noexcept
...
+Vector Misc::calculateRelativeAngle(const Vector& source, const Vector& destination) noexcept
+{
+    Vector delta = destination - source;
+    Vector angles{ radiansToDegrees(atan2f(-delta.z, std::hypotf(delta.x, delta.y))), radiansToDegrees(atan2f(delta.y, delta.x)) };
+    angles.normalize();
+    return angles;
+}
+
+//Some distance check function i found in this source i think
+int Misc::distance(Vector a, Vector b) noexcept
```
Comment (Laxybaxy) [inline Osiris/Hacks/Misc.cpp]: remove

File: Osiris/Hacks/Misc.h [hunk 4]
```diff
@@ -37,6 +38,9 @@ namespace Misc
     void fakePrime() noexcept;
     void killMessage(GameEvent& event) noexcept;
 
+    Vector calculateRelativeAngle(const Vector& source, const Vector& destination) noexcept;
+    int distance(Vector a, Vector b) noexcept;
```
Comment (Laxybaxy) [inline Osiris/Hacks/Misc.h]: remove

**Comment by NekoRem** [comment]
Yeah it was easier sorry should have given credit

**Comment by Evilhelder** [comment]
One question: What would a code look like to calculate and show on the screen the damage caused by "Team Damage" by the allies in the match?

**Comment by Evilhelder** [comment]
P.S .: I'm still working on a way to make an Auto Run boost code.

**Comment by xAkiraMiura** [comment]
> One question: What would a code look like to calculate and show on the screen the damage caused by "Team Damage" by the allies in the match?

how is that on topic to a block bot?

**Comment by NekoRem** [comment]
> > One question: What would a code look like to calculate and show on the screen the damage caused by "Team Damage" by the allies in the match?
> 
> how is that on topic to a block bot?

indeed...

**Comment by Evilhelder** [comment]
sorry guys... I'm stupid.

**Comment by giangero1** [comment]
@NekoRem Check this code out maybe you can add it on osiris it's for crouch blockbot and runboost u know.

void blockbot_crosshair(sdk::user_cmd* cmd) {
	//Return if we are moving
	if (cmd->sidemove != 0 || cmd->forwardmove != 0)
		return;

	sdk::base_entity* local = interfaces::client_entity_list->get_client_entity(intefaces::engine->get_local_player());
	sdk::base_entity* closest_teammate = nullptr;

	if (!local || !local->alive())
		return;

	//Only target teammates within 245 units
	float dist = 245.f;
	for (int i = 0; i < interfaces::engine->max_clients(); i++) {
		sdk::base_entity* current_ent = interfaces::client_entity_list->get_client_entity(i);

		if (!current_ent || current_ent == local)
			continue;

		if (!current_ent->alive() || current_ent->dormant() || current_ent->team() != local->team())
			continue;

		float current_dist = local->abs_origin().dist(current_ent->abs_origin());
		if (current_dist < dist) {
			dist = current_dist;
			closest_teammate = current_ent;
		}
	}

	//Do we have a teammate we can block?
	if (!closest_teammate)
		return;

	//Distance we want to keep from the player
	float radius = 45.f;

	static auto pi = std::atan(1) * 4;
	auto target_origin = closest_teammate->abs_origin();
	auto target_view = closest_teammate->view_angles().normalized().y;

	//Normalize the view to 0-360Β°
	if (target_view < 0.f)
		target_view += 360.f;

	//Get the sin of our view in radians (used for y position)
	auto sin_view = sin(target_view / 180.f * pi);

	//Get the lenght of the adjacent side
	auto delta_x = sqrt(1 - sin_view * sin_view) * radius;

	/*
		Because a square root can be +- something, we need to account for negative values
		Don't know why but we need to negate the distance between 90Β° and 270Β°
	*/
	if (target_view > 90.f && target_view < 270.f)
		delta_x = -delta_x;

	//This is our target position now
	target_origin.x += delta_x;
	target_origin.y += sin_view * radius;

	//"InTeRpOlAtE"
	target_origin += closest_teammate->velocity().length_2d() * interfaces::global_vars->interval_per_tick;

	//Get our head position
	auto head_pos = local->hitbox_position(sdk::hitbox::head);

	//Null z pos
	auto teammate_origin = closest_teammate->abs_origin();
	head_pos.z = teammate_origin.z = 0.f;

	//Return if we aren't close enough
	if (teammate_origin.dist(head_pos) > 245.f)
		return;

	//Move towards our point
	auto dist_x = head_pos.x - target_origin.x;
	auto dist_y = head_pos.y - target_origin.y;

	cmd->sidemove = dist_x > 0 ? 450.f : -450.f;
	cmd->forwardmove = dist_y > 0 ? 450.f : -450.f;

	//Fix our movement (we applied it from a y = -90 perspective
	//				   Old angles, Current cmd
	util.movement_fix(sdk::vector(0, -90, 0), cmd);
}

**Comment by gmDope** [comment]
errors
can somebody help me fast ? pls

**Comment by NekoRem** [comment]
You’re missing the config stuff

**Comment by rmccrystal** [comment]
@danielkrupinski this seems nice to have.

**Comment by NekoRem** [comment]
Yeah but its not good code i might upload better version of this

**Comment by Evilhelder** [comment]
Sad :( 
image

**Comment by NekoRem** [comment]
Not interested anymore.

**Comment by newcommerdontblame** [comment]
can u remake clear pr for this?

**Comment by NekoRem** [comment]
> can u remake clear pr for this?

sure why not

**Comment by aljawary** [comment]
does it still working?

**Comment by ekstazas** [comment]
any plans to update this?

**Comment by NekoRem** [comment]
> any plans to update this?

As i said i will soon i do have other things in life to do

**Pull request closed without merging**